src/components/Planning/ProjectProgrammedDialog/ProjectProgrammedDialog.tsx

```diff
@@ -0,0 +1,152 @@
+    setProjectsForSubmit((current) =>
+      current.filter((p) => {
+        return p.label !== projectName;
+      }),
+    );
+  }, []);
+
+  const { Header, Content, ActionButtons } = Dialog;
+
+  const handleDialogClose = useCallback(() => {
+    setProjectsForSubmit((current) => ({ ...current, projectsForSubmit: [] }));
+    handleClose();
+  }, [handleClose]);
+
+  const onSubmit = useCallback(
+    async (e: MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      dispatch(silentPatchProjectsThunk(buildRequestPayload(projectsForSubmit))).then(() => {
+        setProjectsForSubmit((current) => ({
+          ...current,
+          projectsForSubmit: [],
+        }));
+      });
+    },
+
+    [dispatch, buildRequestPayload, projectsForSubmit],
+  );
+
+  return (
+    <div>
+      {/* Dialog */}
+      <div className="display-flex-col">
+        <Dialog
+          id="add-project-programmed-dialog"
+          aria-labelledby={'add-project-programmed-dialog-label'}
+          isOpen={isOpen}
+          close={handleDialogClose}
+          closeButtonLabelText={t('closeGroupFormWindow')}
+          className="big-dialog"
+          scrollable
+        >
+          {/* Header */}
+          <Header id={'add-project-programmed-header'} title={t(`addProjectsToProgramming`)} />
+
+          <Content>
+            <div className="dialog-search-section">
+              <div>
+                <ProjectProgrammedSearch
+                  onProjectClick={onProjectClick}
+                  onProjectSelectionDelete={onProjectSelectionDelete}
+                  projectsForSubmit={projectsForSubmit}
+                />
+              </div>
+            </div>
+          </Content>
+          <ActionButtons>
+            <Button
+              onClick={onSubmit}
+              disabled={!(projectsForSubmit && projectsForSubmit.length > 0)}
+              data-testid="search-projects-button"
+            >
+              {t('search')}
+            </Button>
+            <Button onClick={handleDialogClose} variant="secondary" data-testid="cancel-search">
+              {t('cancel')}
+            </Button>
+          </ActionButtons>
+        </Dialog>
+      </div>
+    </div>
+  );
+};
+
+const ProjectProgrammedDialog: FC = () => {
+  const selectedClass = useAppSelector(selectSelectedClass);
+  const selectedSubClass = useAppSelector(selectSelectedSubClass);
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleSetOpen = useCallback(() => setIsOpen((current) => !current), []);
+  const onOpenGroupForm = useCallback(
+    (e: MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      handleSetOpen();
+    },
+    [handleSetOpen],
+  );
+  const handleClose = useCallback(() => {
+    handleSetOpen();
+  }, [handleSetOpen]);
+  return (
+    <div>
+      {isOpen && <DialogContainer isOpen={isOpen} handleClose={handleClose} />}
```

Comment:
is it check for isOpen necessary here? Will dialog component within dialog container take care of its own visibility?

Reply:
done